packages/apollo-engine-reporting/src/plugin.ts

```diff
@@ -78,6 +79,14 @@ export const plugin = <TContext>(
        */
       const logger = requestLogger || loggerForPlugin;
 
+      // If the options are false don't do any metrics timing.
```

Comment:
Can we add a guard-rail here for non-TypeScript users?

```suggestion
      if(!['function', 'boolean', 'undefined'].includes(
        typeof options.traceReporting,
      )) {
        throw new Error("Invalid option passed to `traceReporting`.");
      };

      // If the options are false don't do any metrics timing.
```